### DISCUSSION TITLE
Any tips for speeding up Llama 2 models run entirely on CPU (not GPU)?

### DISCUSSION THREAD

**Discussion by CalculonPrime** [oobabooga/text-generation-webui#4097] [Q&A]
I have a system with lots of cores and 128GB of DDR5.  I can run a 70b model but I only get 0.6-.8 tokens/second.  Any tips for improving performance without any GPU involvement?  Flags I should set?  I installed and applied xformers, but didn't help much.  (I gather now that xformers is only for GPUs.)

If I could reach even 2 tokens/second, I think that would be a great improvement, but I'm not sure what optimizations would be required for that.

**Comment by Tedy50** [comment]
I think it is very dependent on memory bandwidth wasu more than on processing power because entire model will need to be processed  for every token along with context.

You can try to optimize by selecting best number of threads, because as I noticed more threads may ruin performance
